Source/Core/Common/HttpRequest.cpp

```diff
@@ -24,23 +25,26 @@ class HttpRequest::Impl final
     POST,
   };
 
-  explicit Impl(std::chrono::milliseconds timeout_ms);
+  explicit Impl(std::chrono::milliseconds timeout_ms, ProgressCallback callback);
 
   bool IsValid() const;
   Response Fetch(const std::string& url, Method method, const Headers& headers, const u8* payload,
                  size_t size);
 
+  static int CurlProgressCallback(Impl* impl, double down, double up, double, double);
```

Comment:
Could you keep the same parameter names (`double dlnow, double dltotal, double ulnow, double ultotal`) for consistency?